### PR TITLE
feat(dashboard,core-flows,js-sdk,types): ability to mark payment as paid

### PR DESCRIPTION
**Pull request by riqwan**
what:

- adds ability to mark payment collection as paid

https://github.com/user-attachments/assets/4297db53-5638-4ec9-b084-9faadc2995aa

### CODE REVIEW

File: packages/core/core-flows/src/order/workflows/mark-payment-collection-as-paid.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    const payment = authorizePaymentSessionStep({
+      id: paymentSession.id,
+      context: { order_id: input.order_id },
+    })
+
+    capturePaymentWorkflow.runAsStep({
+      input: {
+        payment_id: payment.id,
+        captured_by: "req.auth_context.actor_id",
```
Comment (carlos-r-l-rodrigues) [inline packages/core/core-flows/src/order/workflows/mark-payment-collection-as-paid.ts]: it should come from the input?

**Review by carlos-r-l-rodrigues — APPROVED** [review]

File: packages/core/core-flows/src/order/workflows/mark-payment-collection-as-paid.ts (hunk 1 above)
Comment (riqwan) [reply packages/core/core-flows/src/order/workflows/mark-payment-collection-as-paid.ts]: good catch

Comment (riqwan) [reply packages/core/core-flows/src/order/workflows/mark-payment-collection-as-paid.ts]: pushed

**Comment by olivermrbl** [comment]
You can only mark system payments as paid, correct?

**Comment by riqwan** [comment]
@olivermrbl yes, thats correct. Did I miss something?

File: packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-summary-section/order-summary-section.tsx [hunk 2]
```diff
@@ -104,10 +110,54 @@ export const OrderSummarySection = ({ order }: OrderSummarySectionProps) => {
     (pc) => pc.status === "not_paid"
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-summary-section/order-summary-section.tsx:110]: question: Can we rely on this check? couldn't there be multiple unpaid collections?

**Comment by olivermrbl** [comment]
> @olivermrbl yes, thats correct. Did I miss something?

No no, I am just trying to wrap my head around how this will work with other payment collections on the order 😄

**Comment by riqwan** [comment]
> > @olivermrbl yes, thats correct. Did I miss something?
> 
> No no, I am just trying to wrap my head around how this will work with other payment collections on the order 😄

There's no way to get in this situation through the UI as of yet. So not possible right now. But when we link the payment collection to the order change, we can get rid of this check

File: packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-summary-section/order-summary-section.tsx (hunk 2 above)
Comment (riqwan) [reply packages/admin-next/dashboard/src/routes/orders/order-detail/components/order-summary-section/order-summary-section.tsx:110]: There's no way to get in this situation through the UI as of yet. So not possible right now. But when we link the payment collection to the order change, we can get rid of this check

**Pull request merged**